sonar-plugin-api/src/main/java/org/sonar/api/batch/fs/internal/DefaultInputFile.java

```diff
@@ -49,7 +50,7 @@
   private int lastValidOffset = -1;
   private String hash;
   private int nonBlankLines;
-  private int[] originalLineOffsets;
+  private List<Integer> originalLineOffsets;
```

Comment:
to really improve memory consumption, I'd rather track down the reason why we have a ``List`` of ``Integer`` in the first place as a parameter of constructor of class ``Metadata``.

Apparently, this is because of class `org.sonar.api.batch.fs.internal.FileMetadata.LineOffsetCounter`.

This class receives ``char`` and stores them as Integer. This is the initial cause of boxing.

In addition, the internal ``List`` is created with the default size, which might not be pertinent. We might be consuming to much memory here.

But at the end of the day, having an internal array of ``int`` would require a bit of code (dealing with expending the array with a array copy when required) but would not be so hard with a real memory benefit.